### PR TITLE
Sending WhatsApp Messages with Serverless Python Applications using AWS Chalice

### PR DESCRIPTION
**Pull request by J-rayX**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [X] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by J-rayX** [comment]
This fixes #2287

**Comment by ahmadmardeni1** [comment]
👋

**Comment by lalith1403** [comment]
This is an interesting article. Yet to perform a tech review, but since it involves multiple steps, the probability of the reader getting lost is high. You should add screenshots at the end of every stage in such tutorials to navigate the reader better.

**Comment by J-rayX** [comment]
Hi @lalith1403,
Thanks for the review and corrections.

1. I noticed you switched the level 2 headers to level 3 headers. I'd love to know why. Thanks.

2. **About adding images to all the section:**
Actually, not all the sections can have images. Some have, the others have terminal outputs rather which are shown in the article like the following: 

- Introduction
- Prerequisites 
- Setting up Dependencies (image)
- Setting up the Database (terminal output)
- Sending WhatsApp message
- Deploy to AWS (images)
- Conclusion

So, not all the sections seem appropriate with images. Let me know what you think?

**Comment by lalith1403** [comment]
Perfect. If you feel it adds sufficient value, then I am all in with that decision. Regarding the first point, refer to the [contributing guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/example_article_submission.md). It is a formatting style we use to publish articles. All headers are H3 and beyond.

File: articles/aws-chalice-twilio-whatsapp/index.md [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+images:
+
+  -url: /engineering-education/aws-chalice-twilio-whatsapp/hero.jpg
+   alt: AWS Chalice example image
+---
+
+
+### Introduction:
+
+Serverless computing enables developers to build software and applications without dealing with servers. It abstracts server management from the responsibilities and worries of a developer. AWS Chalice is a light and fast serverless framework built by AWS. It is a Python-based framework. It leverages the Amazon API Gateway and AWS Lambda.
```
Comment (lalith1403) [inline articles/aws-chalice-twilio-whatsapp/index.md:21]: Insert links to WhatsApp APIs, AWS Chalice, Amazon API Gateway, and Lambda. Let there be no ambiguity wrt the products being mentioned.

File: articles/aws-chalice-twilio-whatsapp/index.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+Let's check out the setup. Navigate back to the application folder `welcome-app` to use the `chalice local` command:
+
+```bash
+chalice local
+```
+
+We should get an output like this:
+
+```bash
+Serving on http://127.0.0.1:8000
```
Comment (lalith1403) [inline articles/aws-chalice-twilio-whatsapp/index.md:208]: Let the reader know what to expect on localhost port 8000.

File: articles/aws-chalice-twilio-whatsapp/index.md [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+Serverless computing enables developers to build software and applications without dealing with servers. It abstracts server management from the responsibilities and worries of a developer. AWS Chalice is a light and fast serverless framework built by AWS. It is a Python-based framework. It leverages the Amazon API Gateway and AWS Lambda.
+
+WhatsApp is a free messaging platform used by over 2 billion people across the world. WhatsApp Messaging API allows developers to build applications for WhatsApp users.
+
+In this article, we set up a Chalice application with the DynamoDB database. We will also integrate Twilio WhatsApp messaging and send messages from our application.
+
+### Prerequisites
+
+- Python 3.6 or a later version
+- venv or virtualenv
```
Comment (lalith1403) [inline articles/aws-chalice-twilio-whatsapp/index.md:30]: Insert a link to creating virtual environments using virtualenv. You can also mention the pip command to install the library.

File: articles/aws-chalice-twilio-whatsapp/index.md [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        ReadCapacityUnits: "5"
+        WriteCapacityUnits: "5"
+
+Outputs:
+  TableName:
+    Value: !Ref "chaliceDemo"
+    Description: Name of the newly created DynamoDB table
+```
+
+In the above file, we indicated the key attributes of our DynamoDB table, i.e., the `id` and the `name` of the message recipients. The `KeyShema` consists of the primary key. The [AWS documentation](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html) contains more information on attribute definition in DynamoDB tables.
```
Comment (lalith1403) [inline articles/aws-chalice-twilio-whatsapp/index.md:189]: KeySchema -> KeySchema

**Review by lalith1403 — CHANGES_REQUESTED** [review]
A few minor changes and this should be good to go. 

This is a fascinating article with in-depth technical expertise shown. Kudos to you on writing a masterpiece, and I hope you come up with many more. Congratulations and best of luck!

**Comment by J-rayX** [comment]
Thanks for the review and the positive comments about the article @lalith1403 
I'm encouraged.

I have added the links and the output returned at port 8000.
Once again thank you.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by ahmadmardeni1** [comment]
This PR has the old structure @hectorkambow

**Comment by lalith1403** [comment]
I will ask and guide the author to create a new PR. @ahmadmardeni1

**Comment by lalith1403** [comment]
@J-rayX. The repository that you are using is the earlier one. We recently shifted to a new structure. Can you clone the new repository, paste your article in the new cloned repository and submit a new PR.

**Comment by ahmadmardeni1** [comment]
Please @ me when he opens a new PR, I will do the final review immediately. @lalith1403

**Comment by J-rayX** [comment]
> @J-rayX. The repository that you are using is the earlier one. We recently shifted to a new structure. Can you clone the new repository, paste your article in the new cloned repository and submit a new PR.

where's the new repository at, please, @lalith1403 @ahmadmardeni1?

**Comment by lalith1403** [comment]
Closing this PR. Duplicate PR merged.

**Pull request closed without merging**